rubyforgood/human-essentials
# Prevent import of inventory if storage location already has items in it and double importing

Fixes https://github.com/rubyforgood/human-essentials/issues/3687

Partially borrowed from https://github.com/rubyforgood/human-essentials/pull/3696

### Description
Importing storage location inventory can have 2 weird behaviours
- If imported in two separate tabs of the same storage location, the inventory is doubled when it should be one import. This isn't handled by the current [front end check](https://github.com/rubyforgood/human-essentials/blob/deaeb409372767613a6753f2fd2f6355a9045110/app/views/storage_locations/show.html.erb#L53)
- If `Import CSV` is double clicked, there is double inventory when it should be just one import.

The fix is 
- A back end check that redirects to the page with an error if the storage location already has inventory
- Disabling the `Import CSV` button with a greyed out look and text once clicked.

### Type of change
* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
*Double tab issue*
1. Make sure a storage location has already been setup
2. Navigate to `http://localhost:3000/diaper_bank/storage_locations/<id>` to show an individual storage location in *both* tabs
3. Click `Import Baseline Inventory' in both tabs
4. Upload a sample csv
5. Click `Import CSV` in both tabs
6. In first tab, see the import is successful
7. In second tab, see error message in screenshot is displayed

*Double click issue*
1. Make sure a storage location has already been setup
2. Navigate to `http://localhost:3000/diaper_bank/storage_locations/<id>` to show an individual storage location in one tab
3. Click `Import Baseline Inventory' in 
4. Upload a sample csv
5. Try to double click `Import CSV` in one tab
6. See you can't double click the button as it is disabled and says `Importing` like in screenshot.

### Screenshots
<img width="968" alt="Screenshot 2024-04-23 at 5 25 50 PM" src="https://github.com/rubyforgood/human-essentials/assets/131324731/6578fe82-51f5-46d5-a264-741239a4248b">
<img width="630" alt="Screenshot 2024-04-23 at 5 32 20 PM" src="https://github.com/rubyforgood/human-essentials/assets/131324731/b0694771-cc3d-4acc-9d38-ed45286fdd14">
